### PR TITLE
Dwight/finance tool

### PR DESCRIPTION
**Pull request by dwight-foster**
# Description

This is a tool to connect agents to the yahoo finance. It lets them get access to financials, news, and recommendations for different stocks. I also fixed a type on the Contribution.md file where package was misspelled. 

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [x] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -288,7 +288,7 @@ cd ./llama-index-packs
 llamaindex-cli new-package --kind "packs" --name "my new pack"
 
 cd ./llama-index-integrations/readers
-llamaindex-cli new-pacakge --kind "readers" --name "new reader"
+llamaindex-cli new-package --kind "readers" --name "new reader"
```
Comment (logan-markewich) [inline CONTRIBUTING.md:291]: ha thanks!

File: llama-index-integrations/tools/llama-index-tools-yahoo-finance/llama_index/tools/yahoo_finance/base.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
+from llama_index.core.tools.tool_spec.base import BaseToolSpec
+import yfinance as yf
```
Comment (logan-markewich) [inline llama-index-integrations/tools/llama-index-tools-yahoo-finance/llama_index/tools/yahoo_finance/base.py:2]: can we add this dep to the pyproject.toml? (and can you run `poetry lock` after that and commit the lockfile?

Comment (dwight-foster) [reply llama-index-integrations/tools/llama-index-tools-yahoo-finance/llama_index/tools/yahoo_finance/base.py:2]: Yes just did that.

**Comment by logan-markewich** [comment]
Very cool, thanks!

**Pull request merged**